### PR TITLE
docs: Rename `host` options to `endpoint`

### PR DESCRIPTION
**Pull request by ktff**
Ref. #1775

This PR address the core of the #1775 and renames `host` options to `endpoint` where applicable. Old names are still usable, while the documentation is updated to use `endpoint`.

This also changes:
 * ~~`datadog_logs` sink `endpoint` option into `address` since the option fits address definition and not endpoint.~~ (Edit: reverted)
 * ~~`papertrail` sink `endpoint` option into `address` for the same reason.~~ (Edit: reverted)
 * `pulsar` sink `address` option into `endpoint` since that option fits endpoint definition.

### CODE REVIEW

**Comment by bruceg** [comment]
I don't see why the datadog_logs and papertrail sinks are being moved _away_ from the `endpoint` setting. Do they not also use an URL, as evidenced by the use of `UriSerde`?

**Comment by ktff** [comment]
> why the datadog_logs and papertrail sinks are being moved away from the endpoint setting

`datadog_logs` and papertrail sinks only use host and port part of `UriSerde`, which corresponds to address in other sinks, for example `socket` sink.

Now, that is current situation, but with future upgrades mentioned by @jszwedko it makes more sense to retain them as endpoints. 

> The downside is the schemes would be required even though only one is current supported

We can supply a default scheme if none is specified.

File: .meta/sinks/papertrail.toml.erb [hunk 1]
```diff
@@ -31,9 +31,9 @@ write_to_description = "[Papertrail][urls.papertrail] via [Syslog][urls.papertra
   encodings: ["json", "text"],
 ) %>
 
-[sinks.papertrail.options.endpoint]
+[sinks.papertrail.options.address]
```
Comment (binarylogic) [inline .meta/sinks/papertrail.toml.erb:34]: Should this remain as `endpoint`?

File: .meta/sinks/datadog_logs.toml.erb [hunk 2]
```diff
@@ -37,12 +37,12 @@ examples = ["${DATADOG_API_KEY_ENV_VAR}", "ef8d5de700e7989468166c40fc8a0ccd"]
 required = true
 description = "Datadog [API key](https://docs.datadoghq.com/api/?lang=bash#authentication)"
 
-[sinks.datadog_logs.options.endpoint]
+[sinks.datadog_logs.options.address]
```
Comment (binarylogic) [inline .meta/sinks/datadog_logs.toml.erb:40]: Same here. Should this remain as `endpoint`?

File: .meta/sinks/papertrail.toml.erb (hunk 1 above)
Comment (ktff) [reply .meta/sinks/papertrail.toml.erb:34]: Yes, forgot about that. Good catch.

File: .meta/sinks/datadog_logs.toml.erb (hunk 2 above)
Comment (ktff) [reply .meta/sinks/datadog_logs.toml.erb:40]: This also.

**Pull request merged**